openssl/plan.sh

```diff
@@ -45,10 +45,15 @@ _common_prepare() {
       "$PLAN_CONTEXT/ca-dir.patch" \
       | patch -p1 --backup
 
-  # Purge the codebase (mostly tests) of the hardcoded reliance on `/bin/rm`.
-  grep -lr '/bin/rm' . | while read -r f; do
-    sed -e 's,/bin/rm,rm,g' -i "$f"
-  done
+  # The openssl build process hard codes /bin/rm in many places. Unfortunately
+  # we cannot modify the contents of the build scripts to fix this or else we
+  # risk violating the sanctity of the official fips build process.
+  # Instead, we link rm to maintain integrity.
+  # Reference: https://www.openssl.org/docs/fips/UserGuide-2.0.pdf
+  if [[ ! -f "/bin/rm" ]]; then
+    hab pkg binlink core/coreutils rm
```

Comment:
One correction, this should be 

```
hab pkg binlink core/coreutils /bin/rm
```

Reply:
Ah, I see what you're saying. By default, this will binlink it into /bin/rm, but by being more explicit, we can be more certain that it will continue to work in case Habitat updates the default binlink location to somewhere else.

Reply:
@nellshamrell @echohack you were both inaccurate 😛 

```
[3][default:/src:0]# hab pkg binlink core/coreutils /bin/rm
» Binlinking /bin/rm from core/coreutils into /bin
✗✗✗
✗✗✗ `/bin/rm' was not found under any 'PATH' directories in the core/coreutils/8.25/20170513213226 package
✗✗✗
```

```
[2][default:/src:0]# hab pkg binlink core/coreutils rm --dest /bin/rm
» Binlinking rm from core/coreutils into /bin/rm
Ω Creating parent directory /bin/rm
★ Binlinked rm from core/coreutils/8.25/20170513213226 to /bin/rm/rm
```

Reply:
Doh, it's the destination directory, not the full path.